pex/pip/version.py

```diff
@@ -114,14 +115,29 @@ def __get__(self, obj, objtype=None):
                     self._default = preferred_version
                     break
             if self._default is None:
-                self._default = max(
-                    (
-                        version
-                        for version in PipVersionValue._iter_values()
-                        if not version.hidden and version.requires_python_applies(current_version)
-                    ),
-                    key=lambda pv: pv.version,
+                applicable_versions = tuple(
+                    version
+                    for version in PipVersionValue._iter_values()
+                    if not version.hidden and version.requires_python_applies(current_version)
                 )
+                if not applicable_versions:
+                    raise ValueError(
```

Comment:
When working through `--pip-version` pip24 hiddenness, I was getting a semi-cryptic error message from `max` getting an empty iterable. Erroring was correct, but now the message is more clear in this case.